src/pal/src/misc/sysinfo.cpp

```diff
@@ -525,5 +525,20 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     }
 #endif
 
+#ifdef HAVE_SYSCTL
```

Comment:
This needs to be `#if HAVE_SYSCTL`, cmake always defines a symbol we check for and assigns it a value 0 or 1 based on whether the check succeeded.